### PR TITLE
[en] hyphenated compounds

### PR DESCRIPTION
**Pull request by evan-defran-lt**

### CODE REVIEW

**Comment by danielnaber** [comment]
Could you add a test to `AbstractEnglishSpellerRuleTest`?

**Comment by evan-defran-lt** [comment]
@danielnaber, just added a test. 
Java programming is new to me, so any feedback is greatly appreciated.

File: languagetool-language-modules/en/src/main/java/org/languagetool/rules/en/AbstractEnglishSpellerRule.java [hunk 1]
```diff
@@ -1336,6 +1336,36 @@ protected List<SuggestedReplacement> getAdditionalTopSuggestions(List<SuggestedR
...
+      for (String part : parts) {
+        if (isMisspelled(part)) {
+          List<String> partSuggestions = speller1.getSuggestions(part);
+          if (partSuggestions.size() == 0) {
+            partSuggestions = speller2.getSuggestions(part);
+          }
+          if (partSuggestions.size() > 0) {
+              String suggestion = getHyphenatedWordSuggestion(parts, i, partSuggestions.get(0));
+              topSuggestions.add(new SuggestedReplacement(suggestion));
+            }
```
Comment (danielnaber) [inline languagetool-language-modules/en/src/main/java/org/languagetool/rules/en/AbstractEnglishSpellerRule.java:1351]: indentation is off here

File: languagetool-language-modules/en/src/main/java/org/languagetool/rules/en/AbstractEnglishSpellerRule.java [hunk 2]
```diff
@@ -1336,6 +1336,36 @@ protected List<SuggestedReplacement> getAdditionalTopSuggestions(List<SuggestedR
...
+
+  private String getHyphenatedWordSuggestion(String[] parts, int currentPos, String currentPostSuggestion) {
+    List<String> newParts = new ArrayList<>();
+    for (int j = 0; j < parts.length; j++) {
+        if (currentPos == j) {
+            newParts.add(currentPostSuggestion);
+          } else {
+            newParts.add(parts[j]);
+          }
+      }
```
Comment (danielnaber) [inline languagetool-language-modules/en/src/main/java/org/languagetool/rules/en/AbstractEnglishSpellerRule.java:1365]: indentation if off here

File: languagetool-language-modules/en/src/test/java/org/languagetool/rules/en/AbstractEnglishSpellerRuleTest.java [hunk 3]
```diff
@@ -95,7 +95,14 @@ public void testNonVariantSpecificSuggestions(Rule rule, Language language) thro
     // http://grammar.yourdictionary.com/spelling-and-word-lists/misspelled.html
     // https://en.wikipedia.org/wiki/Commonly_misspelled_English_words#cite_note-YD-4
   }
+  public void testHyphenatedWordSuggestions(Rule rule, Language language) throws IOException {
+    this.lt = new JLanguageTool(language);
+    this.rule = rule;
 
+    assertFirstMatch("long-trem", "long-term");
```
Comment (danielnaber) [inline languagetool-language-modules/en/src/test/java/org/languagetool/rules/en/AbstractEnglishSpellerRuleTest.java:102]: Looks good!

**Review by danielnaber — CHANGES_REQUESTED** [review]

**Pull request merged**